ui/effects/use-debounce.js

```diff
@@ -0,0 +1,18 @@
+// @flow
+import { useEffect, useState } from 'react';
+
+export default function useDebounce(value: string, delay: number) {
```

Comment:
This signature would have been:

```typescript
function useDebounce<T>(value: T, delay: number): T
```

But flow seemed to be doing some wonky stuff and propagating the value oddly.  This is also the reason for the explicit "string" on the `subscriptionFilter` declaration.